api/service/syncing/syncing.go

```diff
@@ -562,3 +554,30 @@ func (ss *StateSync) RegisterNodeInfo() int {
 	}
 	return count
 }
+
+// getMaxPeerHeight gets the maximum blockchain heights from peers
+func (ss *StateSync) getMaxPeerHeight() uint64 {
+	ss.CleanUpNilPeers()
+	maxHeight := uint64(0)
+	for id := range ss.syncConfig.peers {
+		peerConfig := ss.syncConfig.peers[id]
+		response := peerConfig.client.GetBlockChainHeight()
```

Comment:
Better make these calls go routine. Otherwise, it will be slow.

Reply:
I changed to goroutine. Please take a look at the changes